interfaces/builtin/basedeclaration.go

```diff
@@ -441,6 +441,11 @@ slots:
     deny-auto-connection: true
     deny-connection:
       on-classic: false
+  online-accounts-service:
+    allow-installation:
+      slot-snap-type:
+        - app
+    deny-connection: true
```

Comment:
This is in contrast to say, an implicit interface that is provided by core for some service. In that situation, we trust that the service is operating within the interface design constraints because the service is coming from a trusted archive (eg, the Ubuntu archive). With app snap slot implementations, that guarantee is not there so we block the slot implementation by default.